.github/workflows/uc-oss.yml

```diff
@@ -0,0 +1,68 @@
+name: uc-oss
+
+on:
+  pull_request: # for testing
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths:
+      - .github/workflows/uc-oss.yml
+      - mlflow/protos/**
+      - mlflow/store/**
+  schedule:
+    # Run this workflow daily at 13:00 UTC
+    - cron: "0 13 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -exo pipefail {0}
+
+env:
+  MLFLOW_HOME: /home/runner/work/mlflow/mlflow
+
+jobs:
+  uc-oss-integration-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    permissions: {}
+    if: github.event_name == 'workflow_dispatch' || (github.event_name == 'schedule' && github.repository == 'mlflow-automation/mlflow') || (github.event_name == 'pull_request' && github.event.pull_request.draft == false)
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event_name == 'schedule' && 'mlflow/mlflow' || github.event.inputs.repository }}
+          ref: ${{ github.event.inputs.ref }}
+          submodules: recursive
+      - uses: ./.github/actions/free-disk-space
+      - uses: ./.github/actions/setup-python
+      - uses: ./.github/actions/setup-pyenv
+
+      - name: Install dependencies
+        run: |
+          source ./dev/install-common-deps.sh --ml
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin" # Use Temurin distribution of OpenJDK
+
+      - name: Clone UnityCatalog at tag v0.2.1
+        run: |
+          git clone --branch v0.2.1 --depth 1 https://github.com/unitycatalog/unitycatalog.git
+
+      - name: Start uc-oss server
+        working-directory: unitycatalog
+        run: |
+          build/sbt package
+          nohup bin/start-uc-server > uc-server.log 2>&1 &
```

Comment:
I think some form of a wait check (similar to the alive or die util function for the MLflow gunicorn server) is needed for the UC server. Doesn't seem to be a problem since the UC server is started prior to the MLflow server in the existing setup (which should ideally be changed in accordance with Haru's recommendation of starting the server safely with a safe available port and await handling for the server process to be healthy), but this might be a bit hard to troubleshoot if there is some issue with the UC server's main branch if the server fails to respond to requests for some reason.

Reply:
We should only be pulling from the specified release tag and not the main branch.   The goal is to make sure that changes introduced in an mlflow PR aren't the cause of an issue with the integration, so just using the latest stable release of UC OSS is the way to go here.

Reply:
I added the same wait check we do for the mlflow server for the uc-oss server in the build_servers function in the test.